sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c

```diff
@@ -0,0 +1,263 @@
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = ARRAYSIZE(alerts),
+      .classes = classes,
+      .class_configs = class_config,
+      .classes_len = ARRAYSIZE(class_config),
+      .ping_timeout = 0,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+}
+
+/**
+ * Execute the aon timer interrupt test.
+ */
+static void execute_test(dif_aon_timer_t *aon_timer) {
+  uint32_t bark_cycles =
+      aon_timer_testutils_get_aon_cycles_from_us(kWdogBarkMicros);
+  uint32_t bite_cycles =
+      aon_timer_testutils_get_aon_cycles_from_us(kWdogBiteMicros);
+
+  LOG_INFO(
+      "Wdog will bark after %u/%u us/cycles and bite after %u/%u us/cycles",
+      (uint32_t)kWdogBarkMicros, bark_cycles, (uint32_t)kWdogBiteMicros,
+      bite_cycles);
+
+  // Setup the wdog bark and bite timeouts.
+  aon_timer_testutils_watchdog_config(aon_timer, bark_cycles, bite_cycles,
+                                      /*pause_in_sleep=*/false);
+
+  // Trigger the clkmgr fatal error to start escalation.
+  LOG_INFO("Ready for error injection");
+
+  busy_spin_micros(kEscalationPhase0MicrosCpu);
+
+  // This should never run since escalation turns the CPU off.
+  CHECK(false, "The alert handler failed to escalate");
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  init_peripherals();
+
+  // Enable all the interrupts used in this test.
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+  rv_plic_testutils_irq_range_enable(&plic, kPlicTarget,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassd);
+
+  alert_handler_config();
+
+  // Check if there was a HW reset caused by the escalation.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  rst_info = rstmgr_testutils_reason_get();
+  rstmgr_testutils_reason_clear();
+
+  CHECK(rst_info == kDifRstmgrResetInfoPor ||
+            rst_info == kDifRstmgrResetInfoEscalation,
+        "Wrong reset reason %02X", rst_info);
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Booting for the first time, starting test");
+    execute_test(&aon_timer);
+  } else if (rst_info == kDifRstmgrResetInfoEscalation) {
+    LOG_INFO("Booting for the second time due to escalation reset");
+    // Check the clkmgr has no fatal errors due to reset.
+    dif_clkmgr_fatal_err_codes_t codes;
+    CHECK_DIF_OK(dif_clkmgr_fatal_err_code_get_codes(&clkmgr, &codes));
+    CHECK(codes == 0, "Fatal error codes should be cleared upon reset");
```

Comment:
In testplan, I saw it mentioned:
```Upon alert escalation reset, the internal status should be clear and clkmgr should not attempt to send out more alerts.
```
I wonder if we can read the alert_cause register and make sure the value is 0?

Reply:
Good catch! Done.